spec/services/events/stores/aggregated_clickhouse_store_spec.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Events::Stores::AggregatedClickhouseStore, type: :service, clickhouse: true do
+  subject(:event_store) do
+    described_class.new(
+      code:,
+      subscription:,
+      boundaries:,
+      filters: {
+        grouped_by:,
+        grouped_by_values:,
+        charge_id:,
+        charge_filter:,
+        matching_filters:,
+        ignored_filters:
+      }
+    )
+  end
+
+  let(:billable_metric) { create(:billable_metric, field_name: "value", code: "bm:code") }
+  let(:organization) { billable_metric.organization }
+
+  let(:customer) { create(:customer, organization:) }
+  let(:plan) { create(:plan, organization:) }
+  let(:subscription) { create(:subscription, customer:, plan:, started_at:) }
+
+  let(:started_at) { Time.zone.parse("2023-03-15") }
+  let(:code) { billable_metric.code }
+
+  let(:boundaries) do
+    {
+      from_datetime: subscription.started_at.beginning_of_day,
+      to_datetime: subscription.started_at.end_of_month.end_of_day,
+      charges_duration: 31
+    }
+  end
+
+  let(:charge) { create(:standard_charge, plan:, billable_metric:) }
+
+  let(:grouped_by) { nil }
+  let(:grouped_by_values) { nil }
+  let(:with_grouped_by_values) { nil }
+
+  let(:charge_id) { charge.id }
+  let(:charge_filter) { nil }
+  let(:matching_filters) { {} }
+  let(:ignored_filters) { [] }
+
+  let(:events) do
+    events = []
+
+    5.times do |i|
+      properties = {billable_metric.field_name => i + 1}
+      groups = {}
+
+      if i.even?
+        # matching_filters.each { |key, values| properties[key] = values.first }
+
+        applied_grouped_by_values = grouped_by_values || with_grouped_by_values
+
+        if applied_grouped_by_values.present?
+        # applied_grouped_by_values.each { |grouped_by, value| properties[grouped_by] = value }
+        elsif grouped_by.present?
+          grouped_by.each do |group|
+            groups[group] = "#{Faker::Fantasy::Tolkien.character.delete("'")}_#{i}"
+          end
+        end
+      elsif grouped_by.present?
+        grouped_by.each do |group|
+          groups[group] = described_class::NIL_GROUP_VALUE
+        end
+      end
+
+      (ignored_filters.first || {}).each { |key, values| properties[key] = values.first } if i.zero?
+
+      events << Clickhouse::EventsEnrichedExpanded.create!(
+        transaction_id: SecureRandom.uuid,
+        organization_id: organization.id,
+        external_subscription_id: subscription.external_id,
+        subscription_id: subscription.id,
+        plan_id: plan.id,
+        code:,
+        aggregation_type: "count",
+        charge_id:,
+        charge_version: charge.updated_at,
+        charge_filter_id: charge_filter&.id,
```

Comment:
Related to my question above, is the `nil` casted into `""` here ?

Also, I might be wrong, but I believe there's no test where the charge filter id is present.